packages/geo-diary/src/core/private-route-wrapper.tsx

```diff
@@ -36,9 +37,11 @@ export const PrivateRouteWrapper: React.FunctionComponent<PrivateRouteWrapperPro
   hasSession,
 }) => {
   const cognitoClientId = process.env.COGNITO_CLIENT_ID_GEO_DIARY as string
-  const refreshParams = getTokenFromQueryString(location.search, cognitoClientId)
+  const refreshParamsRaw = getTokenFromQueryString(location.search, cognitoClientId)
+  const marketplaceGlobalObject = getMarketplaceGlobalsByKey()
 
-  if (refreshParams && !hasSession) {
+  if (refreshParamsRaw && !hasSession) {
+    const refreshParams: RefreshParams = { ...refreshParamsRaw, mode: marketplaceGlobalObject ? 'DESKTOP' : 'WEB' }
```

Comment:
Just a suggestion but would it not make more sense to add this to getTokenFromQueryString? Then you don't have to repeat yourself. Just a thought...

Reply:
@willmcvay ah great, will check and change if it doesn't affect other things

Reply:
It’s fine as long as you don’t change the returned interface. In this case mode is already returned so will be fine

Reply:
Cool, thanks 💯